### PR TITLE
Remove unused param tpu_core_idx

### PR DESCRIPTION
**Pull request by rohitgr7**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## What does this PR do?
Removed unused parameter `tpu_core_idx`.

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by awaelchli** [comment]
Hi
Did you actually test this?
I'm getting
```
Exception in device=TPU:0: tpu_train() takes 2 positional arguments but 3 were given
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/torch_xla/distributed/xla_multiprocessing.py", line 119, in _start_fn
    fn(gindex, *args)
TypeError: tpu_train() takes 2 positional arguments but 3 were given
```
in the MNIST TPU example
https://colab.research.google.com/drive/1-_LKx4HwAxl5M6xPJmqAAu444LTDQoa3
could we fix this asap, I'd like to test my PR #1756 to make sure TPU works

**Comment by awaelchli** [comment]
Note that we don't have TPU tests yet, so we need to always test it manually in the notebooks.

File: pytorch_lightning/trainer/trainer.py [hunk 1]
```diff
@@ -885,7 +885,7 @@ def fit(
 
             # train
             if self.tpu_id is not None:
-                self.tpu_train(self.tpu_id, model)
+                self.tpu_train(model)
             else:
                 xmp.spawn(self.tpu_train, args=(model,), nprocs=self.tpu_cores, start_method=start_method)
```
Comment (awaelchli) [inline pytorch_lightning/trainer/trainer.py:890]: as you can see in this line, the tpu_train function is passed to this spawn function and it expect the tpu_core_idx.
It's not unused afterall.

Comment (Borda) [reply pytorch_lightning/trainer/trainer.py:890]: dam... we need to get TPU test asap...
cc: @luiscape

**Pull request merged**